qiskit/transpiler/synthesis/aqc/cnot_structures.py

```diff
@@ -0,0 +1,325 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+These are the CNOT structure methods: anything that you need for creating CNOT structures.
+"""
+import logging
+
+import numpy as np
+
+# N O T E: we use 1-based indices like in Matlab. To be updated.
```

Comment:
This off-by-one issue seems to be all over the code and I have the feeling it will be a source of bug in the future. Do you think is worthy to factor that out before merging?

Reply:
Oh, yeah, this was fixed, no more such indexing!